src/d3d9/d3d9_device.cpp

```diff
@@ -5190,25 +5190,58 @@ namespace dxvk {
     // First we calculate the size of that UP buffer slice
     // and store all sizes and offsets into it.
 
+    struct VBOCopy {
+      uint32_t srcOffset;
+      uint32_t dstOffset;
+      uint32_t copyBufferLength;
+      uint32_t copyElementCount;
+      uint32_t copyElementSize;
+      uint32_t copyElementStride;
+    };
     uint32_t upBufferSize = 0;
-    std::array<uint32_t, caps::MaxStreams> vboUPBufferOffsets = {};
-    std::array<uint32_t, caps::MaxStreams> vboUPBufferSizes = {};
-    for (uint32_t i = 0; i < caps::MaxStreams && dynamicSysmemVBOs; i++) {
-      vboUPBufferOffsets[i] = upBufferSize;
+    std::array<VBOCopy, caps::MaxStreams> vboCopies = {};
 
+    for (uint32_t i = 0; i < caps::MaxStreams && dynamicSysmemVBOs; i++) {
```

Comment:
In the other for loop over `caps.MaxStreams` below, the check on `dynamicSysmemVBOs` is done in a preceding `if`, not inside the if condition as happens here. 
Best to be consistent, so either move this loop inside an if block, or update below loop to follow the same approach as here.